alerting/src/test/kotlin/org/opensearch/alerting/resthandler/MonitorRestApiIT.kt

```diff
@@ -1102,4 +1105,89 @@ class MonitorRestApiIT : AlertingRestTestCase() {
+        val monitor = createRandomDocumentMonitor()
+
+        val storedMonitor = getMonitor(monitor.id)
+
+        assertEquals("Indexed and retrieved monitor differ", monitor, storedMonitor)
+    }
+
+    @Throws(Exception::class)
+    fun `test updating conditions for a document level monitor`() {
+        val monitor = createRandomDocumentMonitor()
+
+        val updatedTriggers = listOf(
+            QueryLevelTrigger(
+                name = "foo",
+                severity = "1",
+                condition = Script("return true"),
+                actions = emptyList()
+            )
+        )
+        val updateResponse = OpenSearchRestTestCase.client().makeRequest(
+            "PUT", monitor.relativeUrl(),
+            emptyMap(), monitor.copy(triggers = updatedTriggers).toHttpEntity()
+        )
+
+        OpenSearchRestTestCase.assertEquals("Update monitor failed", RestStatus.OK, updateResponse.restStatus())
+        val responseBody = updateResponse.asMap()
+        OpenSearchRestTestCase.assertEquals("Updated monitor id doesn't match", monitor.id, responseBody["_id"] as String)
+        OpenSearchRestTestCase.assertEquals("Version not incremented", (monitor.version + 1).toInt(), responseBody["_version"] as Int)
+
+        val updatedMonitor = getMonitor(monitor.id)
+        OpenSearchRestTestCase.assertEquals("Monitor trigger not updated", updatedTriggers, updatedMonitor.triggers)
+    }
+
+    @Throws(Exception::class)
+    fun `test deleting a document level monitor`() {
+        val monitor = createRandomDocumentMonitor()
+
+        val deleteResponse = client().makeRequest("DELETE", monitor.relativeUrl())
+        assertEquals("Delete failed", RestStatus.OK, deleteResponse.restStatus())
+
+        val getResponse = client().makeRequest("HEAD", monitor.relativeUrl())
+        assertEquals("Deleted monitor still exists", RestStatus.NOT_FOUND, getResponse.restStatus())
+    }
+
+    @Throws(Exception::class)
+    fun `test creating a document monitor with error trigger`() {
+        val trigger = randomQueryLevelTrigger()
+        val monitor = randomDocumentLevelMonitor(triggers = listOf(trigger))
+        try {
+            client().makeRequest("POST", ALERTING_BASE_URI, emptyMap(), monitor.toHttpEntity())
+            fail("Monitor with illegal trigger should be rejected.")
+        } catch (e: java.lang.Exception) {
+            assertNotEquals(e.message, e.message, "Illegal trigger type, ${QueryLevelTrigger::javaClass.name}, for bucket level monitor")
+        }
+    }
+    @Throws(Exception::class)
+    fun `test creating a query monitor with error trigger`() {
+        val trigger = randomBucketLevelTrigger()
+        val monitor = randomQueryLevelMonitor(triggers = listOf(trigger))
+        try {
+            client().makeRequest("POST", ALERTING_BASE_URI, emptyMap(), monitor.toHttpEntity())
+            fail("Monitor with illegal trigger should be rejected.")
+        } catch (e: java.lang.Exception) {
```

Comment:
Catch `IllegalArgumentException` instead since that is what is supposed to be thrown.

Also for the message if the assert fails is not needed. If you want to add it, replace `e.message` with something stating that a different exception error message was expected.